.github/workflows/split-docker-build.yml

```diff
@@ -0,0 +1,254 @@
+                        type=pep440,pattern={{version}}
+                        type=pep440,pattern={{major}}.{{minor}}
+                        type=pep440,pattern={{major}}
+                        type=sha,prefix=
+                        type=edge
+                    labels: |
+                        org.opencontainers.image.title=resotoshell
+                        org.opencontainers.image.description=Resoto Shell (resh)
+                        org.opencontainers.image.vendor=Some Engineering Inc.
+
+            -   name: Set up QEMU
+                id: qemu
+                uses: docker/setup-qemu-action@v1
+                with:
+                    image: tonistiigi/binfmt:latest
+                    platforms: arm64,amd64
+
+            -   name: Set up Docker Buildx
+                id: buildx
+                uses: docker/setup-buildx-action@v1
+
+            -   name: Cache Docker layers
+                uses: actions/cache@v2
+                with:
+                    path: /tmp/.buildx-cache
+                    key: ${{ runner.os }}-buildx-${{ github.sha }}
+                    restore-keys: |
+                        ${{ runner.os }}-buildx-
+
+            -   name: Log in to Docker Hub
+                if: github.event_name != 'pull_request'
+                uses: docker/login-action@v1
+                with:
+                    username: ${{ secrets.DOCKERHUB_USER }}
+                    password: ${{ secrets.DOCKERHUB_PASS }}
+
+            -   name: Log in to GitHub Container Registry
+                if: github.event_name != 'pull_request'
+                uses: docker/login-action@v1
+                with:
+                    registry: ghcr.io
+                    username: ${{ github.actor }}
+                    password: ${{ secrets.GITHUB_TOKEN }}
+
+            -   name: Build and push resotobase Docker image
+                uses: docker/build-push-action@v2
+                with:
+                    context: .
+                    file: ./Dockerfile.resotobase
+                    platforms: ${{ steps.platform.outputs.targets }}
+                    push: ${{ github.event_name != 'pull_request' }}
+                    build-args: |
+                        SOURCE_COMMIT=${{ github.sha }}
+                        TESTS=false
+                    tags: ${{ steps.basemeta.outputs.tags }}
+                    labels: ${{ steps.basemeta.outputs.labels }}
+                    cache-from: type=local,src=/tmp/.buildx-cache
+                    cache-to: type=local,dest=/tmp/.buildx-cache-new
+
+            -   name: Build and push resotocore Docker image
+                if: github.event_name != 'pull_request'
+                uses: docker/build-push-action@v2
+                with:
+                    context: .
+                    file: ./Dockerfile.resotocore
+                    platforms: ${{ steps.platform.outputs.targets }}
+                    push: ${{ github.event_name != 'pull_request' }}
```

Comment:
This is redundant with the `if` condition specified above.  (Same with the below steps.)

Reply:
Yea I still want to get rid of that `if`. Just not sure how I'll do it exactly. It would be good to ensure these steps succeed, even on PRs. But for that we'd have to store the resotobase image, at least temporarily. I'm thinking of something like on PR storing the baseimage with the full hash as tag and then referencing that. Right now the other Dockerfiles are essentially untested until we merge the PR to main which is not ideal. It's okay for the moment as they are only 3 lines and not very error prone but that might change in the future.

Reply:
We would then need to push the base image to some repository anytime there are changes to a PR, though, if we want to build additional images off of them.  We could maybe create a new private package that isn't visible outside of our org?

Reply:
Why do we need to push the image? Isn't the build cache local and independent of any push to a remote repo?